### PR TITLE
Eventbrite Block: Update saved block markup

### LINKED ISSUE #14506: Eventbrite Block: Default text color of modal button is too dark
#### Steps to reproduce the issue

1. Add an Eventbrite Block while editing a page or post
1. Select the "Button & Modal" embed type in the sidebar

#### What I expected

To be able to see the button text

#### What happened instead

On some themes, the button text defaults to being very dark

**Screenshots**

Screen Shot 2020-01-28 at 19 54 56

### PR DESCRIPTION
**Pull request by Copons**
Fixes #14506

#### Changes proposed in this Pull Request:

* When the Eventbrite uses the Button & Modal embed type, add `.wp-block-button` to the block wrapper in order to appropriately style the `.wp-block-button__link` inside.
* Change the markup to be `<a role="button">` rather than `<button>` to be compatible with WordPress.com `wp_kses` configuration.

| Before | After |
| - | - |
| Screen Shot 2020-01-28 at 19 54 56 | Screenshot 2020-01-29 at 13 50 01 |

⚠️ Note: this is change breaks existing blocks. Since the block is not released yet, I haven't created a deprecation/migration path. Just keep in mind that you might need to let Gutenberg try to automatically fix any existing blocks, or just remove them and add new ones.

**Additional Context**

In Core Guten, `wp-block-button__link` is the inner element inside a `wp-block-button`.
So, it's to be expected that it malfunctions when used "outside" `wp-block-button`.

E.g. this is how a normal Button block is rendered in the editor:
Screenshot 2020-01-29 at 13 31 50

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Bug fix

#### Testing instructions:

* Open the Editor, insert an Eventbrite block, and provide an Eventbrite event URL.
* From the block sidebar, change the Embed Type to "Button & Modal".
* Make sure the previewed button style is consistent to the current theme, instead of being dark grey with black text color.
* Save and view the post.
* Make sure the rendered button is identical to how it looks in the editor.
* Modify the button style from the block sidebar (make it outline, change the colors and the border radius).
* Make sure the button is updated accordingly in both the editor and the front end.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]

**Comment by Copons** [comment]
I've added a new commit that fixes an issue caught by @creativecoder.
On WPCOM, Eventbrite blocks in button mode weren't displayed on the front end.

As it turns out, they were, but stripped of the `<button>` tag, which not only doesn't work, but also sometimes turns the text color the same as the background color, making it invisible.

My best guess is that WPCOM's `kses` strips the `button` tag while Core doesn't, reason why the block worked fine in our local environments.

In 3ecf911 I've simply changed the `RichText` output from `button` to `a`.
It works for me on both Jetpack and Simple, but please double check for possible regressions.

**Comment by Copons** [comment]
For more context:

The original approach of outputting a `button` element when the Eventbrite block is of type "Button & Modal" doesn't work on WPCOM because `button` is not whitelisted in the KSES.
At the same time, an `a` element that pretends to be a `button` is [not very accessible](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Accessibility).

As a middle ground we are adding a `role="button"`, which should be OK, if not for another issue.
[Button roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role) are supposed to be focusable (and anchors are), have an `onClick` handler (and Eventbrite adds it), and also an `onKeyDown` for <kbd>Enter</kbd> and <kbd>Space</kbd> (and Eventbrite **doesn't** add them).

I think it's a minor issue, and not one we are able to solve cleanly.

This said, I'm not an a11y expert by any means, so let's discuss this further!

**Comment by creativecoder** [comment]
I verified that VoiceOver sees the `<a role=button>` element as a button, so I think this has what it needs for screen reader accessibility.

For wai-aria recommended button behavior, I pushed a commit with some inline javascript to add event handlers. It feels old school, but simple enough that it should be okay--curious what you think about this approach @Copons

File: extensions/blocks/eventbrite/save.js [hunk 1]
```diff
@@ -57,15 +59,18 @@ function saveButton( eventId, attributes ) {
...
+		<div className="wp-block-button">
 			<RichText.Content
-				id={ createWidgetId( eventId ) }
-				tagName="button"
 				className={ buttonClasses }
+				id={ createWidgetId( eventId ) }
+				role="button"
 				style={ buttonStyle }
+				tagName="a"
+				href={ url }
```
Comment (Copons) [inline extensions/blocks/eventbrite/save.js]: Nice one @creativecoder, love that you got to remove the `noscript` with such a simple change! ✨

**Comment by Copons** [comment]
@creativecoder I think the workaround is as good as we could pull off, and tests very well for me with and without JS.

I've added a `target="_blank"` to the button, like it was in the `noscript` version, and cleaned up the spacing of the inline scripts.

**Review by creativecoder — APPROVED** [review]
@Copons I merged in master so I could test the latest version of the block and made a couple of small tweaks.

This looks good, working for me expected from my local Jetpack site and from wpcom, I think it's ready to go.

**Review by jeherve — APPROVED** [review]
This looks good to me, it tests well! 🚢

**Comment by creativecoder** [comment]
r202396-wpcom

**Comment by jeherve** [comment]
Cherry-picked to `branch-8.2` in 1fa4828d371dc5d525b578313b24f805bfb59f54

**Pull request merged**